tests/playwright/shiny/session/clientdata/app.py

```diff
@@ -0,0 +1,21 @@
+# pyright: reportUnknownMemberType=false, reportUnknownVariableType=false
+import matplotlib.pyplot as plt
+import numpy as np
+
+from shiny.express import input, render, session, ui
+
+with ui.sidebar(open="closed"):
+    ui.input_slider("obs", "Number of observations:", min=0, max=1000, value=500)
+
+
+@render.code
+def clientdatatext():
+    cdata = session.clientdata
+    return "\n".join([f"{name} = {cdata[name]()}" for name in reversed(dir(cdata))])
```

Comment:
```suggestion
def client_data_text():
    client_data = session.clientdata
    return "\n".join([
        f"{name} = {client_data[name]()}" for name in reversed(dir(client_data))
    ])
```

To avoid `CDATA` vibes.

Reply:
We don't get autocomplete for `session.clientdata`, right? So you can use `dir()` to find out the methods but you won't get autocomplete for `session.clientdata.url_*()`. The autocomplete would be helpful but isn't a blocker.

Reply:
Nope, no autocomplete. And agreed, it'd be useful, but I also don't see a low effort way to faithfully type those values (not to mention them easily getting out of sync when shiny.js changes).

Reply:
I guess maybe it'd be worth having a `ClientData` class though so we can at least make them discoverable through the API reference? 

```python
class ClientData(Inputs):
     "TODO: docs here"
    pass
```

Reply:
Actually, hold on, maybe there is a sensible thing to do for autocomplete as well...

Reply:
Yeah I also don't think _all the values_ need to be accessible in autocomplete, but it'd be useful to have the `url_` and other stable data as well

Reply:
Thanks for the nudge here -- [e1e4526](https://github.com/posit-dev/py-shiny/pull/1832/commits/e1e45262c593c8daa7ed57dad8573abfed06c4aa) takes a more typing and documentation friendly approach